### PR TITLE
[CHK-1013] vpos integration

### PR DESCRIPTION
**Pull request by ciuffagianluca**
Integration of vpos in payment gateway client

#### List of Changes

Added vpos invocation to transaction authorization request handler

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/main/java/it/pagopa/transactions/client/PaymentGatewayClient.java [hunk 1]
```diff
@@ -131,6 +133,67 @@ public Mono<XPayAuthResponseEntityDto> requestXPayAuthorization(AuthorizationReq
...
+                                                authorizationData.transaction().getTransactionId().value().toString()
+                                        )
+                                        .amount(grandTotal)
+                                        .emailCh(authorizationData.transaction().getEmail().value())
+                                        .holder(cardData.getHolderName())
+                                        .securitycode(cardData.getCvv())
+                                        .isFirstPayment(true) // TODO TO BE CHECKED
+                                        .threeDsData("threeDsData") // TODO From client
+                                        .circuit("VISA") // TODO Add field to card details
+                                        .reqRefNumber("reqRefNumber") // TODO From client
```
Comment (infantesimone) [inline src/main/java/it/pagopa/transactions/client/PaymentGatewayClient.java:165]: to delete

File: src/main/java/it/pagopa/transactions/client/PaymentGatewayClient.java [hunk 2]
```diff
@@ -131,6 +133,67 @@ public Mono<XPayAuthResponseEntityDto> requestXPayAuthorization(AuthorizationReq
...
+                                        encodeMdcFields(authorizationData)
+                                )
+                                .onErrorMap(
+                                        WebClientResponseException.class,
+                                        exception -> switch (exception.getStatusCode()) {
+                                        case UNAUTHORIZED -> new AlreadyProcessedException(
+                                                authorizationData.transaction().getTransactionId()
+                                        ); // 401
+                                        case INTERNAL_SERVER_ERROR -> new BadGatewayException(""); // 500
+                                        default -> exception;
```
Comment (infantesimone) [inline src/main/java/it/pagopa/transactions/client/PaymentGatewayClient.java:191]: 502?

File: src/main/java/it/pagopa/transactions/client/PaymentGatewayClient.java [hunk 3]
```diff
@@ -131,6 +133,67 @@ public Mono<XPayAuthResponseEntityDto> requestXPayAuthorization(AuthorizationReq
...
+                                        "Cannot perform VPOS authorization for null input CreditCardAuthRequestDto"
+                                )
+                        );
+                    }
+                    return creditCardAuthRequest;
+                })
+                .flatMap(
+                        creditCardAuthRequestDto -> creditCardInternalApiClient
+                                .step0CreditCard(
+                                        "Web",
```
Comment (infantesimone) [inline src/main/java/it/pagopa/transactions/client/PaymentGatewayClient.java:180]: the clientId does not need to be set, but is generated by the apim based on the subscription key. If the autogenerated code requires it, you need to fix the pgs openapi.
https://raw.githubusercontent.com/pagopa/pagopa-infra/main/src/core/api/payment_transactions_gateway_api/internal/v1/_openapi.json.tpl

**Review by infantesimone — CHANGES_REQUESTED** [review]

File: src/main/java/it/pagopa/transactions/client/PaymentGatewayClient.java (hunk 1 above)
Comment (ciuffagianluca) [reply src/main/java/it/pagopa/transactions/client/PaymentGatewayClient.java:165]: Thank you

File: src/main/java/it/pagopa/transactions/client/PaymentGatewayClient.java (hunk 2 above)
Comment (ciuffagianluca) [reply src/main/java/it/pagopa/transactions/client/PaymentGatewayClient.java:191]: This are the listed error in the documentation

     * 400 - Bad request - missing mandatory parameters
     * 401 - transactionId already processed
     * 500 - generic error

**Review by infantesimone — APPROVED** [review]

**Review by infantesimone — APPROVED** [review]
🚀

**Pull request merged**